src/seasideimport.h

```diff
@@ -46,7 +55,22 @@ class CONTACTCACHE_EXPORT SeasideImport
     ~SeasideImport();
 
 public:
-    static QList<QContact> buildImportContacts(const QList<QVersitDocument> &details, int *newCounti = 0, int *updatedCount = 0);
+    static QContactFilter localContactFilter();
+    static QList<QContact> buildMergeImportContacts(const QList<QVersitDocument> &details,
+                                                    int *newCount = 0,
+                                                    int *updatedCount = 0);
+    static QList<QContact> buildImportContacts(const QList<QVersitDocument> &details,
+                                               int *newCount = 0,
+                                               int *updatedCount = 0,
+                                               const QSet<QContactDetail::DetailType> &unimportableDetailTypes = (QSet<QContactDetail::DetailType>() << QContactDetail::TypeGlobalPresence << QContactDetail::TypeVersion),
+                                               const QStringList &importableSyncTargets = (QStringList() << QLatin1String("was_local") << QLatin1String("bluetooth")),
+                                               const QContactFilter &mergeMatchFilter = localContactFilter(),
+                                               QContactManager *manager = 0,
+                                               QVersitContactHandler *propertyHandler = 0,
+                                               bool mergeImportListDuplicates = true,
+                                               bool mergeDatabaseDuplicates = true,
+                                               QMap<int, int> *importDuplicateIndexes = 0,
+                                               QMap<int, QContactId> *databaseDuplicateIndexes = 0);
```

Comment:
This is not a good way to go.  Instead, I would suggest adding a ContactBuilder class, and passing an instance of that (or a derivative) to this function.  The passed-in object can handle all the defaults, and can provide a place for the caller to implement tests such as 'bool ContactBuilder::detailImportable(const QContactDetail &)' that can be overridden as required.